src/classes/CRLP_RollupProcessor_TEST.cls

```diff
@@ -302,6 +302,218 @@ private class CRLP_RollupProcessor_TEST {
+        CMT_UnitTestData_TEST.mockFullSetOfRollupDefinitions();
+
+        Account account = buildMockAccount();
+        List<SObject> opps = new List<SObject>{ buildMockOpportunity(account.Id, 100) };
+
+        CRLP_RollupProcessor processor = new CRLP_RollupProcessor()
+            .withBatchJobMode(CRLP_RollupProcessingOptions.BatchJobMode.NonSkewMode)
+            .withRollupType(CRLP_RollupProcessingOptions.RollupType.AccountHardCredit)
+            .withSummaryRecord(account)
+            .withDetailRecords(opps)
+            .withRollupDefinitions(new List<CRLP_Rollup>{ new CRLP_Rollup( new Rollup__mdt() ) });
+
+        CRLP_ApiService_TEST.Stub apiServiceStub = new CRLP_ApiService_TEST.Stub();
+        processor.apiService = (CRLP_ApiService) Test.createStub(CRLP_ApiService.class, apiServiceStub);
+
+        Test.startTest();
+        processor.startRollupProcessing();
+        Test.stopTest();
+
+        System.assert(!apiServiceStub.calledMethods.contains('getBaseRollupStateForRecords'),
+            'Should NOT have attempted to get external rollups'
+        );
+    }
+
+    /**
+     * @description Confirms external rollup data is included in calculations
+     */
+    @isTest
+    private static void shouldIncludeExternalRollupDataInCalculations() {
+        CMT_UnitTestData_TEST.mockFullSetOfRollupDefinitions();
+
+        Account account = buildMockAccount();
+        List<SObject> opps = new List<SObject>{ buildMockOpportunity(account.Id, 100) };
+
+        Test.startTest();
+
+        CRLP_RollupProcessor processor = new CRLP_RollupProcessor()
+            .withBatchJobMode(CRLP_RollupProcessingOptions.BatchJobMode.NonSkewMode)
+            .withRollupType(CRLP_RollupProcessingOptions.RollupType.AccountHardCredit)
+            .withSummaryRecord(account)
+            .withDetailRecords(opps);
+
+        Map<Id, List<CRLP_Rollup>> mockRollupsByParentId = new Map<Id, List<CRLP_Rollup>>{
+            account.Id => buildMockCRLPRollup()
+        };
+
+        CRLP_ApiService_TEST.Stub apiServiceStub = new CRLP_ApiService_TEST.Stub();
+        apiServiceStub.withReturnValue('getBaseRollupStateForRecords', mockRollupsByParentId);
+
+        processor.apiService = (CRLP_ApiService) Test.createStub(CRLP_ApiService.class, apiServiceStub);
+
+        Account updatedAcct = (Account)processor.startRollupProcessing()[0];
+        System.assertEquals(11, updatedAcct.npo02__NumberOfClosedOpps__c,
+            'The number of closed opportunities for this Account should be 11, which includes the 10 from external data');
+        System.assertEquals(1100, updatedAcct.npo02__TotalOppAmount__c,
+            'The total of closed opportunities for this Account should be 1100, which includes the 1000 from external data');
+    }
+
+    /**
+     * @description Builds a mock List<CRLP_Rollup> to be used as for external rollup data
+     */
+    private static List<CRLP_Rollup> buildMockCRLPRollup() {
+        String mockRollupJSON =
```

Comment:
Is there a better way to create this list?

Reply:
Definitely, plus this is fragile (and is in fact no longer valid as per PR 4232). The best way might be to use `CMT_UnitTestData_TEST.mockRollupCmtForAccountHC()` to mock a set of `Rollup__mdt`records for Account Hard Credit. Then use the `CRLP_ApiService.getRollupDefinitions()` method to convert that into `List<CRLP_Rollup>`.

Reply:
I updated the method to use the CRLP_Processor to create some mock Rollups, I think that should make this more robust